explore/exploration.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+
+#include <stdint.h>
+#include <stdexcept>
+#include <algorithm>
+#include <numeric>
+#include <cstring>
+
+namespace exploration
+{
```

Comment:
I believe we should have a nonzero return value for 'bad range' and always guarantee that if the return value is zero then the distribution is normalized.

Reply:
how should we deal with the return values in cb_explore_adf/cb_explore. Throw a VW exception?